src/arch/time.c

```diff
@@ -17,12 +16,37 @@ _v; \
 })
 
 #define CMOS_READ(addr) ({ \
-    outb_p(0x80|addr,0x70); \
+    outb_p(addr,0x70); \
     inb_p(0x71); \
 })
 
+uint8_t hour, minute, second, registerB, day, month, year;
```

Comment:
Don't declare these as static, they should be declared inside the function.

Reply:
Resolved it.